near-sdk-sim/src/runtime.rs

```diff
@@ -83,6 +85,26 @@ impl GenesisConfig {
         });
         signer
     }
+
+    pub fn add_account(mut self, account_id: AccountId, amount: u128) -> Self {
+        self.state_records.push(StateRecord::Account {
+            account_id,
+            account: Account {
+                amount,
+                locked: 0,
+                storage_usage: 0,
+                code_hash: CryptoHash::default(),
+            },
+        });
+        self
+    }
+
+    pub fn load_state_records(mut self, path: &Path) -> Self {
+        let data = fs::read(path).unwrap();
+        let state: Vec<_> = serde_json::from_slice(&data).unwrap();
+        self.state_records.extend(state);
+        self
+    }
```

Comment:
Unlike the API's in near core, this one is a public function that would be called by the contract developers, so the rules here are different. It makes sense to spend some time polishing the API here. I suggest the following signature:

```
pub fn load_state_records(mut self, path: impl AsRef<Path>) -> io::Result<Self> {
```

* this allows the user to pass a string literal as a path
* it moves unwrap to the call-site, which is especially important for libraries.

Reply:
Is `state_records` the best naming here? Is `StateRecord` a concept which is exposed to the contract developers? Or is it some near-core internal detail? If the latter, it might be better to call this just `load_state` or `with_initial_state`?

Reply:
GenericConfig.state_record is public api of near-sdk-sim, so user of near-sdk-sim should be aware of state_records any way? And a mental difference to me is:
- state is a key-value store of the blockchain state
- state_records is a vector of initial state to populate

And the format we take here is vector of records. Does this fit your mental model?

Reply:
>GenericConfig.state_record is public api of near-sdk-sim,

Ah, right, its a public API of the simulator. Then the naming is indeed right, but the argument should be changed to be `AsRef`

Reply:
Cool, AsRef allows pass &str is great